plugin/pkg/admission/initialresources/hawkular.go

```diff
@@ -17,18 +17,195 @@ limitations under the License.
 package initialresources
 
 import (
+	"crypto/tls"
+	"crypto/x509"
 	"fmt"
+	"github.com/golang/glog"
+	"github.com/hawkular/hawkular-client-go/metrics"
+	"io/ioutil"
+	"k8s.io/kubernetes/pkg/api"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
 	"time"
 
-	"k8s.io/kubernetes/pkg/api"
+	kube_client "k8s.io/kubernetes/pkg/client/unversioned"
+	kubeClientCmd "k8s.io/kubernetes/pkg/client/unversioned/clientcmd"
 )
 
-type hawkularSource struct{}
+type hawkularSource struct {
+	client       *metrics.Client
+	uri          *url.URL
+	useNamespace bool
+	modifiers    []metrics.Modifier
+}
+
+const (
+	containerImageTag string = "container_base_image"
+	descriptorTag     string = "descriptor_name"
+	separator         string = "/"
+
+	defaultServiceAccountFile = "/var/run/secrets/kubernetes.io/serviceaccount/token"
```

Comment:
Yea, I've seen this assumption elsewhere. I added serviceAccountFile param to override this path.